caffe2/contrib/aten/aten_op_template.h

```diff
@@ -214,10 +214,10 @@ class ATenOp : public Operator<Context> {
     DEFINE_IF(int64, Long)
     CAFFE_THROW("unsupported type annotation: ", name);
   }
-  at::Type & stringToType(const std::string & name) {
-    return at::getNonVariableType(backend(), stringToScalarType(name));
+  at::TypeExtendedInterface & stringToType(const std::string & name) {
+    return static_cast<at::TypeExtendedInterface&>(at::getNonVariableType(backend(), stringToScalarType(name)));
```

Comment:
why doesn't getNonVariableType / everything in context use and return TypeExtendedInterface?

Reply:
Ok maybe using is hard (unless you de-duplicate legacyTypeDispatch storage), but returning TypeExtendedInterface seems sensible?